library/posture.c

```diff
@@ -21,22 +21,35 @@ limitations under the License.
 #define NANOS(s) ((s) * 1e9)
 #define MILLIS(s) ((s) * 1000)
 
+const int NO_TIMEOUTS = -1;
+
 static void ticker_cb(uv_timer_t *t);
 
 static void ziti_handle_mac(ziti_context ztx, char *id, char **mac_addresses, int num_mac);
 
 static void ziti_handle_domain(ziti_context ztx, char *id, char *domain);
 
-static void ziti_handle_os(ziti_context ztx, char *id, char *os_type, char *os_version, char *os_build);
+static void ziti_handle_os(ziti_context ztx, const char *id, char *os_type, char *os_version, char *os_build);
+
+static void ziti_send_pr(ziti_context ztx);
+
+static void ziti_send_pr_bulk(ziti_context ztx);
+
+static void ziti_send_pr_individually(ziti_context ztx);
 
 static void
-ziti_handle_process(ziti_context ztx, char *id, char *path, bool is_running, char *sha_512_hash, char **signers,
+ziti_handle_process(ziti_context ztx, const char *id, char *path, bool is_running, char *sha_512_hash, char **signers,
                     int num_signers);
 
 extern void posture_init(struct ziti_ctx *ztx, long interval_secs) {
     if (ztx->posture_checks == NULL) {
         NEWP(pc, struct posture_checks);
+
         pc->interval = (double) interval_secs;
+        pc->previous_session_id = NULL;
+        pc->must_send_every_time = true;
+        pc->must_send = false;
+        pc->bulk_supported = false;
```

Comment:
should it be set to `true`?

it is set to false on 404, and I don't see when it becomes `true`

Reply:
That was me testing the individual posting support. It should be `true`

Reply:
Fixed